### PR TITLE
#Task4.2

### PR DESCRIPTION
**Pull request by MEHV29**
This contains the point 4.2 of the thask

### CODE REVIEW

File: Task4/Task4_2/RationalNumber.cs [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Task4_2
+{
+    internal sealed class RationalNumber : IComparable
+    {
+        public int Numerator { get; }
+
```
Comment (IgorKheidorov) [inline Task4/Task4_2/RationalNumber.cs:6]: Why denominator is byte? What about number,  for example, 1/1024?

File: Task4/Task4_2/RationalNumber.cs [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+        int GetLCM(int a, int b)
+        {
+            int gcd = GetGCD(a, b);
+            return Math.Abs(a * b) / gcd;
+        }
+
+        public int CompareTo(object? obj)
+        {
+            if (obj is RationalNumber)
+            {
```
Comment (IgorKheidorov) [inline Task4/Task4_2/RationalNumber.cs:58]: I see your idea to compare via cast to double, this is not a precise approach, and low performance. Try to find other decision

File: Task4/Task4_2/RationalNumber.cs [hunk 3]
```diff
@@ -0,0 +1,147 @@
...
+        }
+
+        public static RationalNumber operator *(RationalNumber number, RationalNumber other)
+        {
+            var numerator = number.Numerator * other.Numerator;
+            var denominator = number.Denominator * other.Denominator;
+
+            return new RationalNumber(numerator, (byte)denominator);
+        }
+
```
Comment (IgorKheidorov) [inline Task4/Task4_2/RationalNumber.cs:128]: check all operand in all operations for null, for "/" check other for 0;

**Review by IgorKheidorov — CHANGES_REQUESTED** [review]
Please fix comments))

File: Task4/Task4_2/RationalNumber.cs (hunk 1 above)
Comment (MEHV29) [reply Task4/Task4_2/RationalNumber.cs:6]: Modified

File: Task4/Task4_2/RationalNumber.cs (hunk 2 above)
Comment (MEHV29) [reply Task4/Task4_2/RationalNumber.cs:58]: Fixed

File: Task4/Task4_2/RationalNumber.cs (hunk 3 above)
Comment (MEHV29) [reply Task4/Task4_2/RationalNumber.cs:128]: Added

**Review by IgorKheidorov — APPROVED** [review]
Cool!

**Pull request merged**